### ISSUE TITLE
Unicode decode error 

### ISSUE THREAD

**Issue by lwesterhof** [irods/python-irodsclient#124]
When querying iRODS for a data object with a name containing diacritics (ǠǡǢǣǤǥǦǧǨǩǪǫǬǭǮǯǰǱǲǳǴǵǶǷǸ.txt) a UnicodeDecodeError is thrown:

```python
self.session.query(DataObject, Collection.name)
                .filter(DataObject.path == phy_path)
                .filter(DataObject.resc_hier == resc_hier)
                .first()
```

```bash
File "/usr/lib/python2.7/site-packages/irods/query.py", line 230, in first
    results = query.execute()
File "/usr/lib/python2.7/site-packages/irods/query.py", line 177, in execute
    message_body = self._message()
File "/usr/lib/python2.7/site-packages/irods/query.py", line 160, in _message
    'InxValPair_PI': self._conds_message()
File "/usr/lib/python2.7/site-packages/irods/query.py", line 138, in _conds_message
    if isinstance(criterion.query_key, Column)
File "/usr/lib/python2.7/site-packages/irods/column.py", line 39, in value
    return self.query_key.column_type.to_irods(self._value)
File "/usr/lib/python2.7/site-packages/irods/column.py", line 115, in to_irods
    return u"'{}'".format(data)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc7 in position 62: ordinal not in range(128)
```

**Comment by trel** [comment]
Used by https://github.com/UtrechtUniversity/irods-consistency-check

**Comment by lwesterhof** [comment]
This is only an issue for Python 2.7, works fine for Python 3. 
Data string should be converted to Unicode first.
https://github.com/irods/python-irodsclient/blob/800488b7b3af97342f229945a8da69c931f09180/irods/column.py#L116 

`data = unicode(data, "utf-8", "replace")`

**Comment by trel** [comment]
To be more precise, this the proposed fix?

```diff
diff --git a/irods/column.py b/irods/column.py
index f4f644f..022f322 100644
--- a/irods/column.py
+++ b/irods/column.py
@@ -113,6 +113,7 @@ class String(ColumnType):
 
     @staticmethod
     def to_irods(data):
+        data = unicode(data, "utf-8", "replace")
         return u"'{}'".format(data)
 
```

As I understand (from https://docs.python.org/2/library/functions.html#unicode), this will not touch unicode objects (here, `data`) in Python3, but in Python2, this will replace any non-unicode characters in `data` with their unicode counterpart, and gracefully handle non-representable characters with `U+FFFD`.

**Comment by lwesterhof** [comment]
Yes, this fixes the problem for me on Python 2.7 and 3.
